### PR TITLE
Fix bug in determining data range

### PR DESCRIPTION
**Pull request by peterfpeterson**
The issue was when a single xmin/xmax was specified for a dataset
with multiple spectra

Description of work.

**To test:**

```python
LoadNexusMonitors(Filename='NOM_91603', OutputWorkspace='NOM_91603')
NormaliseByCurrent(InputWorkspace='NOM_91603', OutputWorkspace='NOM_91603')
ConvertUnits(InputWorkspace='NOM_91603', OutputWorkspace='NOM_91603',
               Target='Wavelength', EMode='Elastic')
ResampleX(InputWorkspace='NOM_91603', OutputWorkspace='NOM_91603',
         XMin=.1, XMax=2.9, NumberBins=6000, LogBinning=True)
```

NOMAD data isn't specifically required, just event data with more than one spectrum that you want to `ResampleX`.

*There is no associated issue.*

**Release Notes** are in the diffraction section, because that is who uses this algorithm.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Algorithms/src/ResampleX.cpp [hunk 1]
```diff
@@ -114,8 +114,23 @@ map<string, string> ResampleX::validateInputs() {
...
-  bool updateXMins = xmins.empty(); // they weren't set
-  bool updateXMaxs = xmaxs.empty(); // they weren't set
+  const size_t numSpectra = inputWS->getNumberHistograms();
+
+  // pad out the ranges by copying the first value to the rest that are needed
+  if (xmins.size() == 1) {
+    const double value = xmins.front();
+    while (xmins.size() < numSpectra)
+      xmins.push_back(value);
+  }
```
Comment (quantumsteve) [inline Framework/Algorithms/src/ResampleX.cpp]: Use std::vector::insert. The third form looks like what you've done here and below 

http://en.cppreference.com/w/cpp/container/vector/insert

File: Framework/Algorithms/src/ResampleX.cpp [hunk 2]
```diff
@@ -114,8 +114,21 @@ map<string, string> ResampleX::validateInputs() {
...
+  // pad out the ranges by copying the first value to the rest that are needed
+  if (xmins.size() == 1 && numSpectra > xmins.size()) {
+    const double value = xmins.front();
+    xmins.insert(xmins.end(), numSpectra - xmins.size(), value);
+  }
+  if (xmaxs.size() == 1 && numSpectra > xmaxs.size()) {
+    const double value = xmaxs.front();
+    xmaxs.insert(xmaxs.end(), numSpectra - xmaxs.size(), value);
+  }
+
```
Comment (quantumsteve) [inline Framework/Algorithms/src/ResampleX.cpp]: Please update the documentation for each parameter.

**Review by quantumsteve — CHANGES_REQUESTED** [review]
I tried the included script and it functions as described in the PR. The documentation should be updated to include this behaviour and/or a log message should be added stating this workaround.

File: Framework/Algorithms/src/ResampleX.cpp (hunk 2 above)
Comment (peterfpeterson) [reply Framework/Algorithms/src/ResampleX.cpp]: The [algorithm documentation](http://docs.mantidproject.org/nightly/algorithms/ResampleX-v1.html) already states this. It was missing behavior that was only recently noticed.

Comment (quantumsteve) [reply Framework/Algorithms/src/ResampleX.cpp]: The fact is was only recently noticed suggests we need to make it more visible in the documentation, such as in the parameter description. At the same time I worry about getting too deeply into the realm of yak shaving.

**Review by quantumsteve — APPROVED** [review]

**Pull request merged**